include/seahorn/BvOpSem2.hh

```diff
@@ -10,6 +11,12 @@
 
 #include <boost/container/flat_set.hpp>
 
+#include "seadsa/Global.hh"
+
+#include "clam/CfgBuilder.hh"
```

Comment:
Add forward declaration instead of including the header file

Reply:
Q: What is the purpose to use forward declaration instead of including header? How could I decide when I need to use forward declaration or header?

Reply:
The forward declaration avoids to include the header file which is a good thing. 

Rather than having 
```
#include "clam/CfgBuilder.hh"
...

std::unique_ptr<clam::CrabBuilderManager> m_cfg_builder_man;
```
You can write

```
namespace clam {
class CrabBuilderManager
}
...
std::unique_ptr<clam::CrabBuilderManager> m_cfg_builder_man;
```

So in the second case, we don't need to include the header file for `CrabBuilderManager`  and the code will still compile. Having less header files, faster compilation times. 

Note that this only works because `m_cfg_builder_man` is a pointer to `CrabBuilderManager`. This is the only information the compiler needs to know when it compiles this file. That is, it will allocate 8 bytes (or whatever the size of std::unique_ptr) to m_cfg_builder_man regardless the type of the pointee.

Reply:
I see. Thanks for the advice.